librocksdb-sys/build.rs

```diff
@@ -224,11 +230,6 @@ fn build_rocksdb() {
         config.define("ROCKSDB_IOURING_PRESENT", Some("1"));
     }
 
-    if env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap() != "64" {
```

Comment:
so removing this changes the behaviour for non android targets. I can't comment on the change for the android targets, but if that's what is failing, let's continue to have this apply for non andorid builds

Reply:
I'm not sure if the code here is correct. Because if "CARGO_CFG_TARGET_POINTER_WIDTH" is NOT equal to 64, then it will define "_FILE_OFFSET_BITS" to 64.

I can keep these lines and filter the Android target here if you think it's necessary.

Reply:
```
    if &target != "armv7-linux-androideabi"
        && env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap() != "64"
    {
        config.define("_FILE_OFFSET_BITS", Some("64"));
        config.define("_LARGEFILE64_SOURCE", Some("1"));
    }
```

Reply:
> I'm not sure if the code here is correct. Because if "CARGO_CFG_TARGET_POINTER_WIDTH" is NOT equal to 64, then it will define "_FILE_OFFSET_BITS" to 64.
> 
> I can keep these lines and filter the Android target here if you think it's necessary.

I looked at https://github.com/rust-rocksdb/rust-rocksdb/pull/754 which added this (and it also apparently broke other builds but that was not addressed?). It's not clear to me why after detecting the pointer width is not 64, we still force it anyway, is this for cross compilation? I'll defer to @aleksuss who may have more context on this